projects/cdk/components/dialog-host/dialog-host.component.ts

```diff
@@ -33,22 +43,42 @@ const isFakeHistoryState = (
     animations: [TUI_PARENT_ANIMATION],
 })
 export class TuiDialogHostComponent<T extends TuiDialog<unknown, unknown>> {
-    readonly dialogs$ = combineLatest(this.dialogsByType).pipe(
-        map(allTypesDialogs =>
-            new Array<T>()
-                .concat(...allTypesDialogs)
-                .sort((a, b) => a.createdAt - b.createdAt),
-        ),
-    );
+    public dialogs: readonly T[] = [];
+    public isDialogClosesOnBack = defaultClosesOnBack;
```

Comment:
You can keep initial implementation of this. It is never changing amid change detection cycle, unlike dialogs list, so it will not present that issue.